connector/github/github.go

```diff
@@ -303,11 +358,82 @@ func (c *githubConnector) user(ctx context.Context, client *http.Client) (user,
 	return u, nil
 }
 
+// userInOrgTeams first queries the GitHub API for a users' org membership, and
+// if user is an org member, queries the API for team membership within the org.
+// userInOrgTeams will return false if user is not in the org, and an empty
+// slice if user is in the org but not any teams in the org
+//
+// The HTTP passed client is expected to be constructed by the golang.org/x/oauth2 package,
+// which inserts a bearer token as part of the request.
+func (c *githubConnector) userInOrgTeams(ctx context.Context, client *http.Client, userName string, org Org) ([]string, bool, error) {
+	if inOrg, err := c.userInOrg(ctx, client, userName, org.Name); err != nil || !inOrg {
+		return nil, false, err
+	}
+
+	userTeams, err := c.teams(ctx, client, org.Name)
+	if err != nil {
+		return nil, true, err
+	}
+	// Return all teams user is a member of if no team restrictions
+	if len(org.Teams) == 0 {
+		return userTeams, true, nil
+	}
+
+	wantedTeams := make(map[string]struct{})
```

Comment:
`wantedTeams` is probably a bad name here. Can probably just write this as a method

```
func filterTeams(whiteList, userTeams []string) []string
```